zip-bolt-support.rst

```diff
@@ -0,0 +1,408 @@
+(4) Ability to do relative time locks for commitment transactions to support unilateral channel closing.
+(5) Ability to do absolute and relative time locks to support multi-hop payments.
+(6) Ability to validate Bolt-specific commitment opening message and closing signature:
+
+- check the validity of the commitment opening
+- check the validity of randomized/blinded signature on the wallet commitment in closure token
+- check the validity of revocation token signature in the event of a channel dispute by merchant
+ 
+(7) Ability to verify the transaction output such that:
+
+- if customer initiated closing, first output pays out to customer with a time lock (to allow merchant to dispute customer balance) and second output pays out to merchant immediately 
+- if merchant initiated closing, a single output that pays the merchant the full balance of the channel with a time lock that allows for customer dispute
+
+**Channel Operation Assumptions.**
+ - Single-funded channel by customer with a minimum fee paid to the merchant.
+ - Either the customer or the merchant can initiate channel closing.
+ - If the customer initiates closing, then the merchant can dispute the closing transaction if they disagrees with the closure token in the closing transaction.
+ - If the merchant initiates closing, the customer has the opportunity to post their own valid closing transaction. In this case, the merchant has an additional opportunity to validate this closing transaction and can dispute if necessary.
+
+1.1 Conditions for Opening Channel 
+-------------
+
+To open a channel, a customer picks a channel-specific public key, commits to an initial wallet, and receives a signature from the merchant (using their long-term keypair) on that wallet. A wallet consists of a wallet-specific public key, a customer balance, and a total channel balance, and is linked to the customer's channel-specific public key. The channel specific public key, initial customer balance, total channel balance, and initial wallet commitment comprise the customer's channel token.
+
+The keypairs used by both the merchant and the customer must support a blind signature scheme.
+
+1.2 Conditions for Closing Channel
+-------------
+
+A customer should be able to close the channel by either opening the initial wallet commitment (if no payments made) or posting a closing token. 
+
+A merchant should be able to close the channel by either posting their closing token or, if the customer posts an outdated version of their closure token (or opens the initial wallet commitment for the channel after one or more payments have been made), a revocation token.
+
+2. Transparent/Shielded Tx: Using T/Z-addresses and Scripting Opcodes
+-------------
+
+We assume the following specific features are present:
+
+(1) ``OP_CLTV`` - absolute lock time
+(2) ``OP_CSV`` - relative lock time
+(3) shielded address support
+(4) 2-of-2 multi-sig transparent address support (via P2SH)
+(5) Transaction non-malleability
```

Comment:
Zcash does not have this for transactions that involve transparent components (due to `scriptSig` malleability), and deploying SegWit is not feasible for NU3 (and unlikely to happen at all). An alternative approach would need to be proposed, specified, and deployed alongside this ZIP.